core/src/main/java/com/linecorp/armeria/client/retry/AbstractRetryingClientBuilder.java

```diff
@@ -39,61 +39,60 @@
 public abstract class AbstractRetryingClientBuilder<O extends Response> {
 
     @Nullable
-    private final RetryRule retryRule;
+    private final RetryConfigBuilder<O> retryConfig;
 
     @Nullable
-    private final RetryRuleWithContent<O> retryRuleWithContent;
-
-    private int maxTotalAttempts = Flags.defaultMaxTotalAttempts();
-    private long responseTimeoutMillisForEachAttempt = Flags.defaultResponseTimeoutMillis();
+    private final RetryConfigMapping<O> mapping;
 
     /**
-     * Creates a new builder with the specified {@link RetryRule}.
+     * Creates a new builder with the specified {@link RetryConfig}.
      */
-    AbstractRetryingClientBuilder(RetryRule retryRule) {
-        this(requireNonNull(retryRule, "retryRule"), null);
+    AbstractRetryingClientBuilder(RetryConfig<O> retryConfig) {
+        this.retryConfig = requireNonNull(retryConfig, "retryConfig").toBuilder();
```

Comment:
How about just `retryConfigBuilder` for the field name?

Reply:
Done